certification/internal/shell/skopeo.go

```diff
@@ -13,13 +14,24 @@ import (
 type SkopeoCLIEngine struct{}
 
 func (e SkopeoCLIEngine) ListTags(image string) (*cli.SkopeoListTagsReport, error) {
-	imageName := strings.Split(image, ":")[0]
+
+	re, err := regexp.Compile(`(?P<Image>[^@:]+)@*:*.+`)
+
+	if err != nil {
+		log.Error("unable to parse the image name", err)
+		log.Debug("error parsing the image name: ", err)
+		log.Trace("failure in attempt to parse the image name to strip tag/digest")
+		return nil, err
+	}
```

Comment:
Much prefer this over MustCompile, thanks!

Reply:
Do you think the regex logic is worth extracting into a function that we can then unit test against a series of different image formats?

Reply:
This is a very good point! I'll isolate the regex stuffs into a function and add the unit tests.